### ISSUE TITLE
acc dropped when continue train

### ISSUE THREAD

**Issue by wangshauitj** [OpenNMT/OpenNMT-py#2324]
I have trained model for 200k steps and stopped it. I want to continue training the model from 200k steps to 400k steps, but the acc decreases significantly.
I'm curious about how to solve it. The following are log and config：

### train log:
![image](https://user-images.githubusercontent.com/85099972/222368958-6c3136d4-d8eb-4441-9191-8b60fa4074ca.png)

### train config:
```
## Where the samples will be written
save_data: run

# Training files
data:
    corpus_1:
        path_src: data/data_v0.4/wrong_address.train.subword
        path_tgt: data/data_v0.4/gt_address.train.subword
        transforms: [filtertoolong]
    valid:
        path_src: data/data_v0.4/wrong_address.dev.subword
        path_tgt: data/data_v0.4/gt_address.dev.subword
        transforms: [filtertoolong]

# Vocabulary files, generated by onmt_build_vocab
src_vocab: run/run_v0.4/source.vocab
tgt_vocab: run/run_v0.4/target.vocab

# Vocabulary size - should be the same as in sentence piece
src_vocab_size: 50000
tgt_vocab_size: 50000

# Filter out source/target longer than n if [filtertoolong] enabled
src_seq_length: 150
src_seq_length: 150

# Tokenization options
src_subword_model: data/data_v0.3/source.model
tgt_subword_model: data/data_v0.3/target.model

# Where to save the log file and the output models/checkpoints
log_file: log/train_v0.4_log
save_model: models/model_v0.4/model.fren

# Stop training if it does not imporve after n validations
early_stopping: 8

# Default: 5000 - Save a model checkpoint for each n
save_checkpoint_steps: 10000

# To save space, limit checkpoints to last n
# keep_checkpoint: 3

seed: 3435

# Default: 100000 - Train the model to max n steps
# Increase to 200000 or more for large datasets
# For fine-tuning, add up the required steps to the original steps
train_steps: 200000

# Default: 10000 - Run validation after n steps
valid_steps: 10000

# Default: 4000 - for large datasets, try up to 8000
warmup_steps: 8000
report_every: 100

# Number of GPUs, and IDs of GPUs
world_size: 2
gpu_ranks: [0, 1]

# Batching
bucket_size: 262144
num_workers: 0  # Default: 2, set to 0 when RAM out of memory
batch_type: "tokens"
batch_size: 4096   # Tokens per batch, change when CUDA out of memory
valid_batch_size: 2048
max_generator_batches: 2
accum_count: [4]
accum_steps: [0]

# Optimization
model_dtype: "fp16"
optim: "adam"
learning_rate: 2
# warmup_steps: 8000
decay_method: "noam"
adam_beta2: 0.998
max_grad_norm: 0
label_smoothing: 0.1
param_init: 0
param_init_glorot: true
normalization: "tokens"
# Model
encoder_type: transformer
decoder_type: transformer
position_encoding: true
enc_layers: 4
dec_layers: 4
heads: 8
hidden_size: 512
word_vec_size: 512
transformer_ff: 2048
dropout_steps: [0]
dropout: [0.1]
attention_dropout: [0.1]
```

### continue train log:
![image](https://user-images.githubusercontent.com/85099972/222382557-662759b6-200b-401f-8fa2-a796613ada98.png)

### continue train config:
```
## Where the samples will be written
save_data: run

# Training files
data:
    corpus_1:
        path_src: data/data_v0.4/wrong_address.train.subword
        path_tgt: data/data_v0.4/gt_address.train.subword
        weight: 7
        transforms: [filtertoolong]
    valid:
        path_src: data/data_v0.4/wrong_address.dev.subword
        path_tgt: data/data_v0.4/gt_address.dev.subword
        transforms: [filtertoolong]

# Vocabulary files, generated by onmt_build_vocab
src_vocab: run/run_v0.4/source.vocab
tgt_vocab: run/run_v0.4/target.vocab

# Vocabulary size - should be the same as in sentence piece
src_vocab_size: 50000
tgt_vocab_size: 50000

# Filter out source/target longer than n if [filtertoolong] enabled
src_seq_length: 150
src_seq_length: 150

# Tokenization options
src_subword_model: data/data_v0.3/source.model
tgt_subword_model: data/data_v0.3/target.model

# Where to save the log file and the output models/checkpoints
log_file: log/train_v0.4.1.1_log
save_model: models/model_v0.4/model.fren

# Stop training if it does not imporve after n validations
early_stopping: 8

# Default: 5000 - Save a model checkpoint for each n
save_checkpoint_steps: 10000

# To save space, limit checkpoints to last n
# keep_checkpoint: 3

seed: 3435

# Default: 100000 - Train the model to max n steps
# Increase to 200000 or more for large datasets
# For fine-tuning, add up the required steps to the original steps
train_from: models/model_v0.4/model.fren_step_200000.pt
train_steps: 400000
reset_optim: none

# Default: 10000 - Run validation after n steps
valid_steps: 10000

# Default: 4000 - for large datasets, try up to 8000
warmup_steps: 8000
report_every: 100

# Number of GPUs, and IDs of GPUs
world_size: 1
gpu_ranks: [0]

# Batching
bucket_size: 262144
num_workers: 0  # Default: 2, set to 0 when RAM out of memory
batch_type: "tokens"
batch_size: 4096   # Tokens per batch, change when CUDA out of memory
valid_batch_size: 2048
max_generator_batches: 2
accum_count: [4]
accum_steps: [0]

# Optimization
model_dtype: "fp16"
optim: "adam"
learning_rate: 2
# warmup_steps: 8000
decay_method: "noam"
adam_beta2: 0.998
max_grad_norm: 0
label_smoothing: 0.1
param_init: 0
param_init_glorot: true
normalization: "tokens"
# Model
encoder_type: transformer
decoder_type: transformer
position_encoding: true
enc_layers: 4
dec_layers: 4
heads: 8
hidden_size: 512
word_vec_size: 512
transformer_ff: 2048
dropout_steps: [0]
dropout: [0.1]
attention_dropout: [0.1]
```

**Comment by vince62s** [comment]
not sure what your task is since your accuracy is really high but maybe it's just overfitting on the beg of your dataset . Are your data shuffled ?

**Comment by wangshauitj** [comment]
1）shuffle data ：I only shuffled the data when making the dataset.

2）Task：My training task is spelling correction.

I analyzed the effect of the model and it seems that it may indeed be overfitting.
I'm not sure if overfitting will cause the acc to drop during continuous training
I wonder if increasing the number of encoding and decoding layers to 6, and increasing the ‘warmup_steps’ would help in this case.
thanks for your answer～
